ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java

```diff
@@ -111,12 +111,15 @@ protected String tableName(Table t) {
     return Warehouse.getQualifiedName(t);
   }
 
-  public static void initializeAndStartThread(CompactorThread thread,
-      Configuration conf) throws Exception {
-    LOG.info("Starting compactor thread of type " + thread.getClass().getName());
-    thread.setConf(conf);
-    thread.init(new AtomicBoolean());
-    thread.start();
+  public static void initializeAndStartThread(CompactorThread thread, Configuration conf) {
+    try {
+      LOG.info("Starting compactor thread of type " + thread.getClass().getName());
+      thread.setConf(conf);
+      thread.init(new AtomicBoolean());
+      thread.start(); 
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
why is this change? is there some compaction related runtime exception we could use?

Reply:
This is called only from `HiveServer2#maybeStartCompactorThreads`, the method that I changed in this PR. 
It is called from within `customPools.forEach(...)` and checked exceptions in functional code are problematic in Java. In reality, a checked exception can never be thrown in this `CompactorThread#initializeAndStartThread` method. It's declared in `CompactorThread#init`, but it is not thrown there. It is done only because classes that extend `CompactorThread` throw exceptions from the `init` method.

Reply:
could we remove then throw from `CompactorThread#initializeAndStartThread/init`?

Reply:
I was wrong saying that the exception can never be thrown from here because this method is called with the Worker object type which can throw exceptions in the init method. So I added a new specialized exception class that extends RuntimeException and used it here.

Reply:
ok, but could we wrap with try-catch only init, not the whole method body

Reply:
Yes, done.